marklogic-client-api-functionaltests/src/test/java/com/marklogic/client/datamovement/functionaltests/ApplyTransformTest.java

```diff
@@ -749,25 +691,29 @@ public void stopTransformJobTest() throws Exception {
 					while (page.hasNext()) {
 						DocumentRecord rec = page.next();
 						rec.getContent(dh);
-						if (dh.get().getElementsByTagName("foo")
-								.item(0).getAttributes().item(0) == null) {
-							count.incrementAndGet();
+						if (dh.get().getElementsByTagName("foo").item(0).getAttributes().item(0) == null) {
+							notTransformedCount.incrementAndGet();
 						}
-
 					}
-					
 				});
 		dmManager.startJob(resultBatcher);				
 		resultBatcher.awaitCompletion();
 
-		System.out.println("stopTransformJobTest: Success: " + successBatch.size());
-		System.out.println("stopTransformJobTest: Skipped: " + skippedBatch.size());
-		System.out.println("stopTransformJobTest : count " + count);
-		// TODO This has been failing intermittently with the sum being 1999 instead of 2000. Interestingly, the
-		// stopJobTransform test in QueryBatcherJobReportTest sometimes fails because it's off by one as well.
-		Assert.assertEquals(2000, successBatch.size() + skippedBatch.size() + count.get());
-		Assert.assertEquals(2000 - count.get(), successBatch.size());
-
+		int successCount = successUris.size();
+		int skippedCount = skippedUris.size();
+		int failedCount = failedUris.size();
+
+		System.out.println("SUCCESS: " + successCount);
+		System.out.println("SKIPPED: " + skippedCount);
+		System.out.println("FAILED: " + failedCount);
+		System.out.println("NOT TRANSFORMED: " + notTransformedCount.get());
+
+		assertEquals(
+			"Unexpected count; success: " + successCount + "; skipped: " + skippedCount + "; failed: " + failedCount,
+			DOC_COUNT, successCount + skippedCount + failedCount + notTransformedCount.get());
```

Comment:
@anu3990 This is the key change here (the rest was just tidying up) - the test now accounts for URIs that failed due to the InterruptedException.